airbyte-integrations/connectors/source-weather/source_weather/source.py

```diff
@@ -193,6 +196,25 @@ def check_connection(self, logger, config) -> Tuple[bool, any]:
         :param logger:  logger object
         :return Tuple[bool, any]: (True, None) if the input config can be used to connect to the API successfully, (False, error) otherwise.
         """
+        if "lat" not in config or "lon" not in config and "apikey" not in config:
+            log("Input config must contain the properties 'api_key' and 'lat'"and 'lon')
+            sys.exit(1)
```

Comment:
You don't need this logic
your config object is automatically validated by your spec.json